### DISCUSSION TITLE
Nemo - swap two selected files

### DISCUSSION THREAD

**Discussion by HT-7** [linuxmint/discussions#117] [Ideas]
Currently, the file names of two files can only be swapped in three steps, by giving one of the files a temporary name.

A "swap file names" shortcut for two selected files which does this process automatically would be convenient.

The temporary file name needs to be checked against a possible existing file name to eliminate the possibility of a file name conflict during the rename process.

**Comment by Secret-chest** [comment]
I second this.

**Comment by rcalixte** [comment]
You could accomplish this with a [Nemo Action and script](https://github.com/linuxmint/nemo/blob/master/files/usr/share/nemo/actions/sample.nemo_action). You could use something like a timestamp and/or special characters as the temporary file name to reduce the potential for collisions.
